synapse/replication/tcp/client.py

```diff
@@ -402,7 +402,7 @@ async def _on_new_receipts(
             if not self._is_mine_id(receipt.user_id):
                 continue
             if (
-                receipt.data.get("hidden", False)
+                receipt.receipt_type == ReceiptTypes.READ_PRIVATE
                 and self._hs.config.experimental.msc2285_enabled
```

Comment:
I think we would have already rejected a private read receipt by the time we get here, so I wonder if the check for the experimental config is necessary?